File: app/src/main/java/com/ninhttd/devtest/utils/NetworkUtils.java
package com.ninhttd.devtest.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.ninhttd.devtest.TekoApplication;

public class NetworkUtils {
    public static boolean isNetworkAvailable() {
        ConnectivityManager cm = (ConnectivityManager) TekoApplication.getAppContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = cm.getActiveNetworkInfo();
        return networkInfo != null && networkInfo.isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/ninhttd/devtest/utils/NetworkUtils.kt
package com.ninhttd.devtest.utils

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

import com.ninhttd.devtest.TekoApplication

object NetworkUtils {
    val isNetworkAvailable: Boolean
        get() {
            val cm = TekoApplication.appContext!!.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val networkInfo = cm.activeNetworkInfo
            return networkInfo != null && networkInfo.isConnectedOrConnecting
        }
}
